### ISSUE TITLE
How to get capture time, exposure time and gain for every frame?

### ISSUE THREAD

**Issue by RadekZenkl** [TheImagingSource/tiscamera#463]
Hi, 
We are using a DFK23UX249 in a fast changing lighting conditions. 
Currently, we record images in python with Gstreamer for later analysis. We would be very interested in knowing the used capture time, exposure time and gain for individual frame. 

Is the capture time metadata working with python bindings for gstreamer? We are trying to extract it with this minimal example:

```
import time
import gi
import numpy as np

gi.require_version("Gst", "1.0")
gi.require_version("GstApp", "1.0")

from gi.repository import Gst, GstApp, GLib

_ = GstApp

def on_new_sample(app_sink):

    sample = app_sink.pull_sample()
    caps = sample.get_caps()

    # Get the actual data
    buffer = sample.get_buffer()
    print(buffer.get_meta("TcamStatisticsMetaApi"))
    print(buffer.pts)

Gst.init(None)

main_loop = GLib.MainLoop()
pipeline = Gst.parse_launch("""tcambin !  video/x-raw,format=BGRx ! appsink sync=true     max-buffers=1 drop=true name=sink emit-signals=true""")
appsink = pipeline.get_by_name("sink")
pipeline.set_state(Gst.State.PLAYING)
handler_id = appsink.connect("new-sample", on_new_sample)

time.sleep(30)

pipeline.set_state(Gst.State.NULL)
main_loop.quit()
```

Is there  a way to get the used exposure time and gain for individual frames? Our approach would be to request camera state, when a new buffer is emmited. Is there a more elegant way to do this?

Thanks

**Comment by RadekZenkl** [comment]
Hi, 
Thank you for your reply. Ok, well then we will try to use that. 

Is the capture time in python still work in progress, or am I doing something wrong.

**Comment by TIS-Stefan** [comment]
> Is the capture time in python still work in progress, or am I doing something wrong.

I am very sorry, but I do not understand your question.

Stefan

**Comment by RadekZenkl** [comment]
There is a sample code for c on how to extract the metadata (https://github.com/TheImagingSource/tiscamera/blob/master/examples/c/10-metadata.c).
However the same example for python is missing. To my understanding, the code from above should do the same thing but it does not return anything. My question is then whether this is because the metadata extraction with python is not possible at the moment, or there is something wrong with the code?

**Comment by TIS-Stefan** [comment]
That is currently not supported in Python, if you use tiscamera 0.14. It is available in tiscamera 1.0: https://github.com/TheImagingSource/tiscamera/blob/development/examples/python/10-metadata.py

You can check out the "development" branch. 

Stefan

**Comment by TIS-Edgar** [comment]
Closed due to inactivity.
